shared/wallets/wallet/container.js

```diff
@@ -36,29 +37,18 @@ const mergeProps = (stateProps, dispatchProps) => {
   sections.push({data: assets, title: 'Your assets'})
 
   // split into pending & history
-  let pending
-  let history
   let mostRecentID
-  stateProps.payments &&
-    stateProps.payments.forEach(p => {
-      if (p.section === 'history') {
-        if (!history) {
-          history = []
-        }
-        history.push({paymentID: p.id, timestamp: p.time})
-      } else if (p.section === 'pending') {
-        if (!pending) {
-          pending = []
-        }
-        pending.push({paymentID: p.id, timestamp: p.time})
-      }
-    })
-  if (history) {
+  const paymentsList = stateProps.payments && stateProps.payments.toList().toArray()
```

Comment:
can't it just be
```
const mapped = (stateProps.payments?.toList().toArray())?.map(p => ({paymentID: p.id, timestamp: p.time}))
const [history, pending] = partition(mapped, p => p.section === 'history')
```

Reply:
Then you'd need to add `section` to the map too

Reply:
ok we can keep it as-is but it could be something like. cause splitPayments should never be null and you don't need to check the param to partition
```
const paymentToItem = p => ({paymentID: p.id, timestamp: p.time})
const [history, pending] = partition(stateProps.payments?.toList().toArray(), p => p.section === 'history').map(arr => arr.map(paymentToItem)
```

Reply:
or more like your existing code
```
  const [_history, _pending] = partition(paymentsList, p => p.section === 'history')
  const mapItem = p => ({paymentID: p.id, timestamp: p.time})
  let history = _history.map(mapItem)
  const pending =_pending.map(mapItem)
```

Reply:
@chrisnojima done